src/main/java/com/rocketden/main/game_object/Game.java

```diff
@@ -24,4 +24,7 @@
 
     private GameTimer gameTimer;
 
+    // Boolean to hold whether all users have solved the problem.
+    private Boolean allSolved = false;
```

Comment:
Thoughts on changing this to something like `isGameOver` and also setting it to true when the GameTimer `timeUp` is hit? That way we only need to check one variable for whether the game is over esp. if we add more end-game conditions in the future - although is there any situation where we would need to distinguish between game over by all solved or by time up?

Reply:
I think in the future, we may have different game modes -- for example, users compete to get the fastest solution, or one that uses the least memory -- which means we'll want to distinguish between `timeUp` and `allSolved`. It may also be useful for players viewing their past games, or for us curious about statistics, to see if the game ended by everybody solving the problem, or by the timer running out.

Reply:
Good points, I hadn't though of that - this works for me